clmm/clusterensemble.py

```diff
@@ -0,0 +1,235 @@
+    id_dict: dict
+        Dictionary of indicies given the cluster id
+    """
+    def __init__(self, unique_id, gclist, *args, **kwargs):
+        """Initializes a ClusterEnsemble object
+        Parameters
+        ----------
+        unique_id : int or string
+            Unique identifier of the galaxy cluster ensemble
+        """
+        if isinstance(unique_id, (int, str)):
+            unique_id = str(unique_id)
+        else:
+            raise TypeError('unique_id incorrect type: %s'%type(unique_id))
+        self.unique_id = unique_id
+        self.data = GCData(meta={'bin_units': None})
+        if len(args)>0 or len(kwargs)>0:
+            self._add_values(gclist, **kwargs)
+
+    def _add_values(self, gc_list, **kwargs):
+        """Add values for all attributes
+
+        Parameters
+        ----------
+        gc_list : list, tuple
+            List of GalaxyCluster objects.
+        gc_cols : list, tuple
+            List of GalaxyCluster objects.
+        """
+        for gc in gc_list:
+            self.make_individual_radial_profile(gc, **kwargs)
+
+    def __getitem__(self, item):
+        """Returns self.data[item]"""
+        return self.data[item]
+
+    def __len__(self):
+        """Returns length of ClusterEnsemble"""
+        return len(self.data)
+
+    def make_individual_radial_profile(self, galaxycluster, bin_units, bins=10, error_model='ste',
+                                       cosmo=None, tan_component_in='et', cross_component_in='ex',
+                                       tan_component_out='gt', cross_component_out='gx',
+                                       tan_component_in_err=None, cross_component_in_err=None,
+                                       weights_in='w_ls', weights_out='W_l'):
+        """Compute the individual shear profile from a single GalaxyCluster object
+        and adds the averaged data in the data attribute.
+
+        Parameters
+        ----------
+        galaxycluster : GalaxyCluster
+            GalaxyCluster object with cluster metadata and background galaxy data
+        bin_units : str
+            Units to use for the radial bins of the shear profile
+            Allowed Options = ["radians", "deg", "arcmin", "arcsec", "kpc", "Mpc"]
+            (letter case independent)
+        bins : array_like, optional
+            User defined bins to use for the shear profile. If a list is provided, use that as
+            the bin edges. If a scalar is provided, create that many equally spaced bins between
+            the minimum and maximum angular separations in bin_units. If nothing is provided,
+            default to 10 equally spaced bins.
+        error_model : str, optional
+            Statistical error model to use for y uncertainties. (letter case independent)
+
+                * `ste` - Standard error [=std/sqrt(n) in unweighted computation] (Default).
+                * `std` - Standard deviation.
+
+        cosmo: dict, optional
```

Comment:
Type would be `CLMM.Cosmology object` I think. Just to be consistent with the docstrings of `util.convert_units`.